examples/replset-docker/README.md

```diff
@@ -0,0 +1,81 @@
+# MongoDB Replication Example Using a Docker
+
+This [MongoDB replication](https://docs.mongodb.com/manual/replication/) example
+uses a [Docker engine](http://docker.com) to run replica set members. This platform
+do not provide any deployment automation and it is used mainly for developing and testing.
+
+## Getting Started
+
+You will need an Docker engine where you can run containers. If you want to avoid running all replset members on one host, you can also use [Docker Swarm](https://docs.docker.com/swarm/).
+
+## Example Working Scenarios
+
+This section describes how this example is designed to work.
+
+### Initial Deployment: 3-member Replica Set
+
+To create replica set with three members you can use this script:
+
+```bash
+function get_ip() {
+    local id="$1" ; shift
+    docker inspect --format='{{.NetworkSettings.IPAddress}}' ${id}
+}
+
+REPLICATION_ARGS="
+-e MONGODB_DATABASE=db
+-e MONGODB_USER=user
+-e MONGODB_PASSWORD=password
+-e MONGODB_ADMIN_PASSWORD=adminPassword
+-e MONGODB_REPLICA_NAME=rs0
+-e MONGODB_KEYFILE_VALUE=xxxxxxxxxxxx
+-e MONGODB_SMALLFILES=true"
+
+docker run -d --name replset0 ${BASE_ARGS} --hostname initialize -e MONGODB_REPLICA_MEMBERS="" $IMAGE_NAME run-mongod-replication
+docker run -d --name replset1 ${BASE_ARGS} -e MONGODB_REPLICA_MEMBERS="$(get_ip replset0)" $IMAGE_NAME run-mongod-replication
+docker run -d --name replset2 ${BASE_ARGS} -e MONGODB_REPLICA_MEMBERS="$(get_ip replset0) $(get_ip replset1)" $IMAGE_NAME run-mongod-replication
+```
+
+`run-mongod-replication` command have to be run in container and in `MONGODB_REPLICA_MEMBERS` environment variable it should contain as much of replica set members addresses as possible.
+
+And later from one of the containers you can also login into MongoDB:
+
+```console
+sh-4.2$ mongo $MONGODB_DATABASE -u $MONGODB_USER -p $MONGODB_PASSWORD --host $MONGODB_REPLICA_NAME/localhost
+MongoDB shell version: 3.2.6
+connecting to: sampledb
+rs0:PRIMARY>
+```
+
+Note: You can also use host version of mongo shell, but you have to substitute values of environmental variables yourself.
+
+During the lifetime of your deployment, one or more of those members might crash or fail. In this case, replica set member is removed. It some special cases, replica set configuration might not be updated -- see Known Limitations.
```

Comment:
> Why is it removed?
> Wouldn't it be a more reasonable workflow that either:
1) set docker settings to automatically restart the container

I was not aware of this. It would be good to mention this in README.

2) operator manually restart the container

It can. Nobody prevents him to do it.

The restarted container probably gets a new IP, so it have to be added to replica set again.

> AFAIK in normal MongoDB deployments, you don't automatically remove members from the replica set config when they go down. Instead, you bring them back up.

In normal MongoDB deployments member IS NOT added to replica set on startup. I've heard many times that deploying in containers is different :-)

I see no difference between some failure of container and terminating of container.